src/helpers/bookmarkModel.js

```diff
@@ -0,0 +1,142 @@
+import Models from '../db/models';
+import ArticleModel from './articles';
+import errorResponse from '.';
+
+const { Article, Bookmark } = Models;
+
+/**
+ * @description class representing Bookmark Helpers
+ * @class BookmarkModel
+ */
+class BookmarkModel {
+  /**
+   * @description - This method is responsible for bookmarking an article
+   * @static
+   * @param {object} response - Response sent from the controller
+   * @param {object} slug - Request sent to the middleware
+   * @param {object} userId - Request sent to the middleware
+   * @returns {object} - object representing newly bookmarked article
+   * @memberof BookmarkModel
+   */
+  static async addBookmark(response, slug, userId) {
+    try {
+      const foundArticle = await ArticleModel.queryForArticle(slug);
+      const articleId = foundArticle.id;
+      await Bookmark.findOrCreate({
+        where: {
+          userId,
+          articleId
+        }
+      }).spread((check, created) => {
+        if (!created) {
+          return response.status(400).json(errorResponse(['Article already bookmarked']));
+        }
+        response.status(200).json({
+          status: 'Success',
+          message: 'Article successfully bookmarked'
+        });
+      });
+    } catch (error) {
+      // logTracker(error);
+      return response.status(500).json(errorResponse(['Could not bookmark article at this time']));
+    }
+  }
+
+  /**
+   * @description - This method is responsible for fetching a user's bookmarks
+   * @static
+   * @param {object} response - Response sent from the controller
+   * @param {object} userId - Request sent to the middleware
+   * @returns {object} - object representing list of bookmarked articles
+   * @memberof BookmarkModel
+   */
+  static async getBookmarks(response, userId) {
+    try {
+      const bookmarks = await Bookmark.findAll({
+        where: {
+          userId
+        },
+        attributes: {
+          exclude: [
+            'userId',
+            'articleId',
+            'createdAt',
+            'updatedAt'
+          ]
+        },
+        include: [
+          {
+            model: Article,
+            attributes: [
+              'title',
+              'description',
+              'imgUrl',
+              'createdAt'
+            ]
+          }
+        ]
+      });
+      if (bookmarks.length < 1) {
```

Comment:
Can `bookmarks.length` possibly be negative? why not use `bookmarks.length === 0`?

Reply:
I have had issues with equality without type coercion in the past, I was comparing two equal numbers and they I was getting errors that they were not equal. So, I have always preferred the less than or equal to sign. Also, length < 1 still scales for when the length is zero, that is why I have always preferred that over using the triple equal sign.

Reply:
Cool. In this case, we're comparing length to a specific number (0), so it can never be a negative.